beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -1457,12 +1410,21 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
     ///
     /// Returns `(block_root, block_slot)`.
     pub fn heads(&self) -> Vec<(Hash256, Slot)> {
-        self.head_tracker.heads()
+        let head_slot = self.canonical_head.cached_head().head_slot();
+        self.canonical_head
+            .fork_choice_read_lock()
+            .proto_array()
+            .viable_heads::<T::EthSpec>(head_slot)
```

Comment:
This `head_slot` is being passed as the `current_slot`, which is not necessarily the same thing. I think we should probably read the slot clock and use its current slot?

Reply:
Head slot is the slot of the head block, which could be potentially quite old. This could affect fork choice calculations related to pulled tips, etc.